benchmarks/run.ml

```diff
@@ -239,107 +258,221 @@ let _ =
+    (match effects with
+    | `None -> ()
+    | _ ->
+        compr_file_size
+          param
+          code
+          Spec.js_of_ocaml_effects
+          sizes.Measure.path
+          (Spec.sub_spec Spec.js_of_ocaml_effects "gzipped");
+        bzip2_file_size
+          param
+          code
+          Spec.js_of_ocaml_effects
+          sizes.Measure.path
+          (Spec.sub_spec Spec.js_of_ocaml_effects "bzip2"));
+    bzip2_file_size
+      param
+      code
+      Spec.js_of_ocaml
+      sizes.Measure.path
+      (Spec.sub_spec Spec.js_of_ocaml "bzip2");
+    runtime_size
+      param
+      code
+      Spec.js_of_ocaml
+      sizes.Measure.path
+      (Spec.sub_spec Spec.js_of_ocaml "runtime");
+    gen_size
+      param
+      code
+      Spec.js_of_ocaml
+      sizes.Measure.path
+      (Spec.sub_spec Spec.js_of_ocaml "generated");
+    gen_size param code Spec.js_of_ocaml_o3 sizes.Measure.path Spec.js_of_ocaml_o3;
+    gen_size
+      param
+      code
+      Spec.js_of_ocaml_js_string
+      sizes.Measure.path
+      Spec.js_of_ocaml_js_string;
+    gen_size param code Spec.js_of_ocaml_inline sizes.Measure.path Spec.js_of_ocaml_inline;
+    gen_size
+      param
+      code
+      Spec.js_of_ocaml_deadcode
+      sizes.Measure.path
+      Spec.js_of_ocaml_deadcode;
+    gen_size
+      param
+      code
+      Spec.js_of_ocaml_compact
+      sizes.Measure.path
+      Spec.js_of_ocaml_compact;
+    gen_size param code Spec.js_of_ocaml_call sizes.Measure.path Spec.js_of_ocaml_call;
+    match effects with
+    | `None -> ()
+    | _ ->
+        gen_size
+          param
+          code
+          Spec.js_of_ocaml_effects
+          sizes.Measure.path
+          Spec.js_of_ocaml_effects);
+  if wasm
+  then (
+    compr_file_size
+      param
+      code
+      Spec.wasm_of_ocaml
+      sizes.Measure.path
+      (Spec.sub_spec Spec.wasm_of_ocaml "gzipped");
+    bzip2_file_size
+      param
+      code
+      Spec.wasm_of_ocaml
+      sizes.Measure.path
+      (Spec.sub_spec Spec.wasm_of_ocaml "bzip2");
+    gen_size
+      param
+      code
+      Spec.wasm_of_ocaml
+      sizes.Measure.path
+      (Spec.sub_spec Spec.wasm_of_ocaml "generated"));
   if compile_only then exit 0;
   Format.eprintf "Measure@.";
   if not nobyteopt
   then (
-    measure param code times Spec.opt "";
-    measure param code times Spec.byte "");
-  let compilers, suites =
+    measure ~param ~code ~meas:times ~spec:Spec.opt "";
+    measure ~param ~code ~meas:times ~spec:Spec.byte "");
+  let interpreters, suites =
     if full
     then
       ( interpreters
-      , [ Some Spec.js_of_ocaml
-        ; Some Spec.js_of_ocaml_o3
-        ; Some Spec.js_of_ocaml_js_string
-        ; Some Spec.js_of_ocaml_unsafe
-        ; Some Spec.js_of_ocaml_inline
-        ; Some Spec.js_of_ocaml_deadcode
-        ; Some Spec.js_of_ocaml_compact
-        ; Some Spec.js_of_ocaml_call
-        ; Some Spec.js_of_ocaml_effects
-        ] )
-    else if effects
```

Comment:
You've dropped this logic. The documentation of `effects` was previously: 
```
only run with and without effect handler support
``` 
which I think meant that we wanted to run a small subset of the runs.